### PR TITLE
Feat/notif updates

### PR DESCRIPTION
**Pull request by themonster2015**
## Description

* Add listener for mutation events and display messages in  the snack bar extension
*  Update notification pages according to the updated design.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/apps/notifications/src/components/pages/customize-notification-page.tsx [hunk 1]
```diff
@@ -218,7 +162,7 @@ const CustomizeNotificationPage: React.FC<CustomizeNotificationPageProps> = ({
     // @TODO: save preferences somewhere, now we just save a key in localstorage to mark as set
     try {
       if (window.localStorage) {
-        localStorage.setItem('notification-preference', JSON.stringify('1')); // @TODO: where to save settings?
+        localStorage.setItem(NOTIF_REF, JSON.stringify(allStates)); // @TODO: where to save settings?
```
Comment (kenshyx) [inline ui/apps/notifications/src/components/pages/customize-notification-page.tsx]: use the `settings` service from the sdk to set these for the current user

File: ui/apps/notifications/src/components/pages/customize-notification-page.tsx [hunk 2]
```diff
@@ -24,93 +23,82 @@ export type CustomizeNotificationPageProps = {
...
+    }
+  }, [allStates]);
 
   const [snoozed, setSnoozed] = React.useState(false);
 
   // check if snooze notification option has already been set
   useEffect(() => {
-    if (window.localStorage) {
+    if (window.localStorage && localStorage.getItem('notifications-snoozed')) {
       setSnoozed(JSON.parse(localStorage.getItem('notifications-snoozed')));
```
Comment (kenshyx) [inline ui/apps/notifications/src/components/pages/customize-notification-page.tsx]: refactor this to use the `settings` service from the sdk

File: ui/apps/notifications/src/components/pages/customize-notification-page.tsx (hunk 1 above)
Comment (kenshyx) [reply ui/apps/notifications/src/components/pages/customize-notification-page.tsx]: the `service` name that you need to specify in the `set` call is the notification app name

File: ui/apps/notifications/src/components/pages/customize-notification-page.tsx [hunk 3]
```diff
@@ -24,93 +23,82 @@ export type CustomizeNotificationPageProps = {
...
-      },
-      {
-        label: t('When there’s new content in topics I’m subscribed to'),
-        value: 'When there’s new content in topics I’m subscribed to',
-        selected: true,
-      },
-    ],
-    [t],
-  );
+  const routing = plugins['@akashaorg/app-routing']?.routing;
```
Comment (didd) [inline ui/apps/notifications/src/components/pages/customize-notification-page.tsx]:
```suggestion
  const routing = getRoutingPlugin();
```

File: ui/apps/notifications/src/extensions/snack-bar-notification.tsx [hunk 4]
```diff
@@ -7,21 +7,90 @@ import {
...
 
+  React.useEffect(() => {
+    if (!message) {
+      setMessageType('success');
+      setMessageUuid('');
+      setAppTitle(null);
+    }
+  }, [message]);
+
+  const routing = plugins['@akashaorg/app-routing']?.routing;
```
Comment (didd) [inline ui/apps/notifications/src/extensions/snack-bar-notification.tsx]:
```suggestion
  const routing = getRoutingPlugin();
```

File: ui/hooks/src/use-settings.ts [hunk 5]
```diff
@@ -63,7 +63,8 @@ export function useSaveSettings() {
 const getSettings = async (app: string) => {
   const sdk = getSDK();
 
-  return sdk.services.settings.get(app);
+  const res = await sdk.services.settings.get(app);
+  return res;
```
Comment (kenshyx) [inline ui/hooks/src/use-settings.ts]: how is this better? O_o

File: ui/apps/notifications/src/components/pages/welcome-page.tsx [hunk 6]
```diff
@@ -33,18 +35,18 @@ const WelcomePage: React.FC<WelcomePageProps> = props => {
...
 
   const navigateTo = getRoutingPlugin().navigateTo;
 
+  const _uiEvents = useRef(uiEvents);
+
   // check if user has gone through onboarding steps before
   let savedPreferences;
   if (window.localStorage) {
-    savedPreferences = JSON.parse(localStorage.getItem('notification-preference'));
+    savedPreferences = JSON.parse(localStorage.getItem(NOTIF_REF));
```
Comment (kenshyx) [inline ui/apps/notifications/src/components/pages/welcome-page.tsx]: these settings should make use of the useSavedSettings hook

File: ui/hooks/src/use-settings.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ * @example useSaveSettings hook
+ * ```typescript
+ * const saveSettings = useSaveSettings();
+ * saveSettings.mutate(JSON.stringify({ app: '@akashaorg/app-akasha-verse', options: [['key', 'value']] }))
+ * ```
+ */
+export function useSaveSettings() {
+  const queryClient = useQueryClient();
+
+  return useMutation((params: string) => saveSettings(params), {
```
Comment (kenshyx) [inline ui/hooks/src/use-settings.ts]: There is no need that the `params` should be a string O_o, you can make use of generics here.
The `app` and `options` properties should be defined as properties of the object param defined here

**Review by kenshyx — APPROVED** [review]

**Pull request merged**